src/Microsoft.TemplateSearch.Common/BlobStorageTemplateInfo.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TemplateEngine.Abstractions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.TemplateSearch.Common
+{
+    [JsonObject(Id = "TemplateInfo")]
+    public class BlobStorageTemplateInfo : ITemplateInfo
+    {
+        [JsonProperty(PropertyName = "Tags")]
+        private IReadOnlyDictionary<string, CacheTag> _tags;
+
+        [JsonProperty(PropertyName = "CacheParameters")]
+        private IReadOnlyDictionary<string, CacheParameter> _cacheParameters;
+
+        [JsonProperty(PropertyName = "BaselineInfo")]
+        private IReadOnlyDictionary<string, BaselineCacheInfo> _baselineInfo;
+
+        private List<ITemplateParameter> _parameters;
+
+        private BlobStorageTemplateInfo() { }
+
+        [JsonIgnore]
+        IReadOnlyList<ITemplateParameter> ITemplateInfo.Parameters
+        {
+            get
+            {
+                if (_parameters == null)
+                {
+                    List<ITemplateParameter> parameters = new List<ITemplateParameter>();
+
+                    foreach (KeyValuePair<string, ICacheTag> tagInfo in Tags)
+                    {
+                        ITemplateParameter param = new TemplateParameter
+                        {
+                            Name = tagInfo.Key,
+                            Documentation = tagInfo.Value.Description,
+                            DefaultValue = tagInfo.Value.DefaultValue,
+                            Choices = tagInfo.Value.Choices,
+                            DataType = "choice"
+                        };
+
+                        if (param is IAllowDefaultIfOptionWithoutValue paramWithNoValueDefault
```

Comment:
We already know the type of `param`. Can we use the type `TeamplateParameter` when declaring `param` and remove the conditional `param is ....`?

Reply:
This would eliminate the cast at line 53.

Reply:
Same comments apply to the foreach below iterating over `CacheParameters`.

Reply:
We should really make this method an extension and stop doing caching of it... and do what you suggested Bekir, I already had fix for this change once...

Reply:
I removed this property and method in my next PR, and kept in only for CLI. I will revise why we needed parameters here and check for alternative.

Reply:
I recalled why it is needed  - some code in CLI depends on parameters, and without it the search tests won't pass. This code will be gone in next PR, so I would like to keep it as if you don't mind. 
In 2 days you will enjoy another nice PR removing it :)